net/connPassError.go

```diff
@@ -0,0 +1,16 @@
+package net
+
+import (
+	"net"
+	"strings"
+)
+
+func passError(err error) bool {
+	if netErr, ok := err.(net.Error); ok && (netErr.Temporary() || netErr.Timeout()) {
+		return true
+	}
+	if strings.Contains(err.Error(), "i/o timeout") {
```

Comment:
If you could reference the code where this error is created, that would help to justify this workaround.